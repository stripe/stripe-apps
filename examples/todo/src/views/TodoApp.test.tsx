import { render, getMockContextProps } from '@stripe/ui-extension-sdk/testing';
import { Button, TextField, List, ListItem } from '@stripe/ui-extension-sdk/ui';
import TodoApp from './TodoApp';

import { ChangeEvent } from 'react';

jest.mock('./stripeClient');

describe('TodoApp', () => {
  // default mock props
  const context = getMockContextProps();

  it('renders the app correctly', async () => {
    const { wrapper, update } = render(<TodoApp {...context} />);

    expect(wrapper).toContainText('Loading...');

    // First wait for the customer to finish loading
    await update();

    expect(wrapper).toMatchSnapshot();
  });
});