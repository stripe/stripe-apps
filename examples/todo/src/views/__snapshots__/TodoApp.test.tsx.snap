// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoApp renders the app correctly 1`] = `
Object {
  "act": [Function],
  "unmount": [Function],
}
`;
